### ISSUE TITLE
Getting Score of Game on Each Play

### ISSUE THREAD

**Issue by stermi01** [BurntSushi/nflgame#274]
Hey guys,

I'm looking to try to get the current score of the game at any point in the game. I.E if we are on a certain play, in the 3rd Quarter with 5 minutes left, I'd like to have the score of the game at that time. I have a theory of where it is, but may someone might be able to help me.

{  
   u'redzone':True,
   u'down':0,
   u'qtr':u'Final',
   u'yl':u'',
   u'clock':   u'00:00   ', u'   media':None,
   u'away':{  
      u'to':0,
      u'score':{  
         u'1':0,
         u'3':8,
         u'2':3,
         u'5':0,
         u'4':10,
         u'T':21

This is the raw data and I think that something like g.away.score.T would be what I need, but I'm not sure how to get it.

If there is another way that'd be great.

**Comment by ochawkeye** [comment]
Afraid there's not an easy way to do that with `nflgame` but you can use `nfldb` to get what you're after quite easily.

```python
import nfldb

db = nfldb.connect()
q = nfldb.Query(db)
q.game(season_year=2016, week=13, season_type='Regular')
start = nfldb.types.Clock.from_str('Q3', '5:00')
for game in q.as_games():
    print game.home_team, 'vs.', game.away_team
    print 'Q3 5:00 score:', game.score_at_time(start)[0], game.score_at_time(start)[1]
    print 'Final score  :', game.home_score, game.away_score
    print '-'*25
```
```
MIN vs. DAL
Q3 5:00 score: 3 7
Final score  : 15 17
-------------------------
ATL vs. KC
Q3 5:00 score: 16 27
Final score  : 28 29
-------------------------
BAL vs. MIA
Q3 5:00 score: 24 0
Final score  : 38 6
-------------------------
CHI vs. SF
Q3 5:00 score: 21 6
Final score  : 26 6
-------------------------
CIN vs. PHI
Q3 5:00 score: 26 0
Final score  : 32 14
-------------------------
GB vs. HOU
Q3 5:00 score: 7 7
Final score  : 21 13
-------------------------
JAC vs. DEN
Q3 5:00 score: 3 17
Final score  : 10 20
-------------------------
NE vs. LA
Q3 5:00 score: 20 3
Final score  : 26 10
-------------------------
NO vs. DET
Q3 5:00 score: 6 16
Final score  : 13 28
-------------------------
OAK vs. BUF
Q3 5:00 score: 16 24
Final score  : 38 24
-------------------------
ARI vs. WAS
Q3 5:00 score: 10 13
Final score  : 31 23
-------------------------
PIT vs. NYG
Q3 5:00 score: 12 7
Final score  : 24 14
-------------------------
SD vs. TB
Q3 5:00 score: 14 10
Final score  : 21 28
-------------------------
SEA vs. CAR
Q3 5:00 score: 30 7
Final score  : 40 7
-------------------------
NYJ vs. IND
Q3 5:00 score: 3 31
Final score  : 10 41
-------------------------
```

**Comment by stermi01** [comment]
Thanks, ya, I'm going to do a workaround then I think

**Comment by ochawkeye** [comment]
I don't believe the source data has the exact information you're looking for.  Here is an example of a play:
```
"477":{
    "sp":0,"qtr":1,"down":1,"time":"08:30","yrdln":"WAS 14","ydstogo":10,"ydsnet":6,"posteam":"NYG",
    "desc":"(8:30) (Shotgun) E.Manning pass short middle to O.Beckham Jr. to WAS 8 for 6 yards (D.Goldson).",
    "note":null,
    "players":{
        "00-0025513":[
            {"sequence":6,"clubcode":"WAS","playerName":"D.Goldson","statId":79,"yards":0}],
        "00-0031235":[
            {"sequence":3,"clubcode":"NYG","playerName":"O.Beckham Jr.","statId":21,"yards":6},
            {"sequence":4,"clubcode":"NYG","playerName":"O.Beckham Jr.","statId":115,"yards":0},
            {"sequence":5,"clubcode":"NYG","playerName":"O.Beckham Jr.","statId":113,"yards":2}],
        "00-0022803":[
            {"sequence":1,"clubcode":"NYG","playerName":"E.Manning","statId":15,"yards":6},
            {"sequence":2,"clubcode":"NYG","playerName":"E.Manning","statId":111,"yards":4}]}},
```
On that play the following players were involved:
* `00-0022803` (Eli Manning) completed a pass (`"statId":15`) for 6 yards.  The pass traveled 4 yards in the air (`"statId":111`)
* `00-0031235` (O.Beckham Jr.) was targeted (`"statId":115`).  It was good for 6 receiving yards (`"statID":21`), of which 2 yards (`"statID":113`) were after the catch
* `00-0025513` (D.Goldson) recorded the tackle (`"statId":79`)

Unfortunately, you won't find anywhere in that listing what the current score is at the time of the play, either before or after.

`nfldb` addresses this by keeping track of scoring plays over the course of the game so it can know at any given point in the game what the current score was.

**Comment by stermi01** [comment]
Appreciated, I ended up doing a workaround in ECL to monitor the score on each play.

**Comment by astromberg** [comment]
stermi01, any chance you can share how you calculated this? I've been looking at trying to get this and it seems that I need to keep track of FG, XP, and TD and total while iterating through the plays. However for turnovers that led to a touchdown, it looks like they are market FUMBLE or INT, not whether or not the team scored.

**Comment by astromberg** [comment]
It looks like maybe I can just lift this logic from nfldb:

https://github.com/BurntSushi/nfldb/blob/9cc2fe8280cada7c1595d54e4dc698b99da72a40/nfldb/types.py#L2347

**Comment by stermi01** [comment]
@astromberg 

I'm using ECL to manage the data once I've pulled it. 

In order to calculate the score I had to calculate the time left in the game on each play, then make sure that each play was sorted by amount of time left in game, finally I iterated through the plays to search for both turnovers and scoring plays within the text and within that Note field as well. The result is that I get an accurate score for each play and added to the respective team that got that scoring play. The only problem is is that on defensive TD plays, there is no way of knowing if they got an XP or not. (Maybe even a 2pt Conv).
